webapp/src/main/java/com/box/l10n/mojito/service/machinetranslation/MachineTranslationService.java

```diff
@@ -0,0 +1,165 @@
+     */
+    @Timed("MachineTranslationService.getLeveragedTranslationResponse")
+    private TranslationsResponse getLeveragedTranslationResponse(List<String> textSources, List<String> targetLocaleTags) {
+        TranslationsResponse translationsResponseDTO = new TranslationsResponse(textSources, targetLocaleTags);
+
+        for (int i = 0; i < textSources.size(); i++) {
+            String sourceText = textSources.get(i);
+            TextUnitTranslationGroup textUnitTranslationGroupDTO = translationsResponseDTO.getTextUnitTranslations().get(i);
+
+            // Note: to further increase leverage performance at a later date, we should introduce a bulk leveraging
+            // function that will take in a list of source strings, a list of target locales and will return one single
+            // result for each source string and locale combination.
+            List<TextUnitDTO> textUnitDTOsForLeveraging = getLeveragedTextUnitDTOS(sourceText);
+
+            if (!textUnitDTOsForLeveraging.isEmpty()) {
+                Comparator<TextUnitDTO> tmTextUnitCreatedDateComparator = Comparator.comparing(TextUnitDTO::getTmTextUnitCreatedDate);
+
+                // Pick the most recent created translations for each language
+                Map<String, TextUnitDTO> mostRecentTranslationsByLanguage = textUnitDTOsForLeveraging.stream()
+                        .collect(Collectors.toMap(TextUnitDTO::getTargetLocale, Function.identity(),
+                                BinaryOperator.maxBy(tmTextUnitCreatedDateComparator)));
+
+                // Copy translations from leveraged matches
+                for (int j = 0; j < textUnitTranslationGroupDTO.getTranslations().size(); j++) {
+                    Translation translation = textUnitTranslationGroupDTO.getTranslations().get(j);
+
+                    // Note: the leveraging mode is locale code specific. If the translation requests ask for "pt" but
+                    // we only have a "pt-BR" translation, or the other way around, we do not use the leveraged match.
+                    if (mostRecentTranslationsByLanguage.containsKey(translation.getLocaleTag())) {
+                        TextUnitDTO matchedLanguageTextUnit = mostRecentTranslationsByLanguage.get(translation.getLocaleTag());
+
+                        translation.setText(matchedLanguageTextUnit.getTarget());
+                        translation.setMatchedTextUnitId(matchedLanguageTextUnit.getTmTextUnitId());
+                        translation.setMatchedTextUnitVariantId(matchedLanguageTextUnit.getTmTextUnitVariantId());
+                        translation.setTranslationSource(TranslationSource.MOJITO_TM_LEVERAGE);
+                    }
+                }
+            } else {
+                logger.debug("No TMTextUnit Match found for content: {}", sourceText);
+            }
+        }
+        return translationsResponseDTO;
+    }
+
+    @SuppressWarnings("SpringCacheableComponentsInspection")
```

Comment:
why is that needed?

Reply:
Due to the caching the results of a non-public method. The error is emitted as that would not generally work - except in our case we use AspectJ - which does ensure that functionality is well covered. (Might make that method public at some point though as there might be some value in it being reused like that).